packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/lib/create_saved_objects.ts

```diff
@@ -74,6 +91,29 @@ export const createSavedObjects = async <T>({
     // the created object if it did not have one (or is omitted if specified)
     const importStateValue = importStateMap.get(`${object.type}:${object.id}`);
     if (importStateValue?.destinationId) {
+      originId = originId ?? object.id;
+
+      // In compatibility mode we generate legacy URL aliases for all objects that have to change ID during import and
+      // origin ID is maintained.
+      const requiresLegacyUrlAlias =
+        compatibilityMode &&
+        originId !== importStateValue.destinationId &&
```

Comment:
__note:__ The `importStateValue.destinationId === originId` is `true` when, for example, we copy object from default space to a custom space, and then copy that object from a custom space back to a default space